### PR TITLE
2 - Instruction

### PR DESCRIPTION
**Pull request by Vryzas**
- introduction to file system package (file read/write)
- sync/async or block/unblock coding types
- create a server commands
- how to create routes
- API (reading/showing data)
- divide an html file into 3 html templates
- filling the html templates with read data
- URL parsing
- npm and packages usage...

### CODE REVIEW

File: 1-node-farm/starter/node_modules/.bin/is-ci [hunk 1]
```diff
@@ -0,0 +1 @@
+../is-ci/bin.js
```
Comment (andre-exd) [inline 1-node-farm/starter/node_modules/.bin/is-ci:1]: Please add a .gitignore and remove node_modules from branch

File: 1-node-farm/starter/templates/template-product.html [hunk 2]
```diff
@@ -32,14 +32,14 @@
       body {
         padding: 5rem 5rem 10rem;
         line-height: 1.7;
-        font-family: 'Nunito Sans', sans-serif;
```
Comment (andre-exd) [inline 1-node-farm/starter/templates/template-product.html:35]: If possible, revert this type of changes

File: 1-node-farm/starter/index.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+const fs = require("fs");
+const http = require(`http`);
+const url = require(`url`);
+// require 3rd party modules
+const slug = require("slugify");
+// require my modules
+const replaceTemplate = require("./modules/replaceTemplate"); // imports the replaceTemplate fn
+const { default: slugify } = require("slugify");
+//////////////////////////////////////////////////////////////////////////////////////
+// // file reading (syncronous or blocking way)
```
[the remark this answers is not in the document]
Comment (Vryzas) [reply 1-node-farm/starter/index.js:12]: Comented code removed

File: 1-node-farm/starter/modules/replaceTemplate.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  let output = temp.replace(/{%PRODUCTNAME%}/g, product.productName);
+  output = output.replace(/{%IMAGE%}/g, product.image);
+  output = output.replace(/{%PRICE%}/g, product.price);
+  output = output.replace(/{%FROM%}/g, product.from);
+  output = output.replace(/{%NUTRIENTS%}/g, product.nutrients);
+  output = output.replace(/{%QUANTITY%}/g, product.quantity);
+  output = output.replace(/{%DESCRIPTION%}/g, product.description);
+  output = output.replace(/{%ID%}/g, product.id);
+
+  if (!product.organic)
```
[the remark this answers is not in the document]
Comment (Vryzas) [reply 1-node-farm/starter/modules/replaceTemplate.js:11]: Added curly braces

File: 1-node-farm/starter/txt/input.txt [hunk 5]
```diff
@@ -1 +1 @@
-The avocado 🥑 is popular in vegetarian cuisine as a substitute for meats in sandwiches and salads because of its high fat content 😄
```
[the remark this answers is not in the document]
Comment (Vryzas) [reply 1-node-farm/starter/txt/input.txt:1]: This text was added automatically by the commented (removed) code as an example on how to write to a file.

File: 1-node-farm/starter/node_modules/.bin/is-ci (hunk 1 above)
Comment (Vryzas) [reply 1-node-farm/starter/node_modules/.bin/is-ci]: Added

File: 1-node-farm/starter/index.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    res.writeHead(200, { "Content-type": `text/html` });
+    const cardsHtml = dataObj
+      .map((el) => replaceTemplate(tempCard, el))
+      .join("");
+    const output = tempOverview.replace("{%PRODUCT_CARDS%}", cardsHtml);
+    // console.log(cardsHtml);
+    res.end(output);
+
+    // Product page
+  } else if (pathname === `/product`) {
```
[the remark this answers is not in the document]
Comment (Vryzas) [reply 1-node-farm/starter/index.js:49]: The code is already correctly formatted.

[the remark this answers is not in the document]
Comment (andre-exd) [reply 1-node-farm/starter/index.js:49]: I see. No need to change then

File: 1-node-farm/starter/txt/input.txt (hunk 5 above)
[the remark this answers is not in the document]
Comment (andre-exd) [reply 1-node-farm/starter/txt/input.txt:1]: I wasn't paying attention. I didn't realised it was an output file. No need for changes

File: 1-node-farm/starter/templates/template-product.html (hunk 2 above)
Comment (andre-exd) [reply 1-node-farm/starter/templates/template-product.html:35]: You haven't reverted these changes on this file

**Review by andre-exd — APPROVED** [review]

**Pull request merged**